private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -69,6 +71,24 @@ type FileInfo interface {
 	isFileInfo()
 }
 
+// DigestChangedError is returned if module pin digests have changed unexpectedly.
+type DigestChangedError struct {
```

Comment:
What's the benefit of doing this public struct style over the `New.../Is...` style that already exists in `storage` and `normalpath`?

Reply:
No real benefit - lemme switch to use that convention.